Source/BlazorStateManager/Mediator/BlazorMediator.cs

```diff
@@ -97,7 +97,7 @@ protected virtual async Task PublishInternal(Type T, string topicString, object
 			foreach (var topic in topicList)
 			{
 				var deadSubscriberList = new Lazy<List<SubscriberInfo>>();
-				foreach (var subscriber in topic.Subscribers)
+				foreach (var subscriber in topic.Subscribers.ToArray())

```

Comment:
Doing a .ToArray() inside of a collection in a foreach is a little unusual and I'd say that it normally requires a comment indicating something like:
// Capturing the current state of the subscribers just in case notifying them causes a change in this collection

Reply:
True, though Phil just merged this so the comment would need to be added later on